src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ImplicitConversion.cs

```diff
@@ -90,6 +91,7 @@ public bool Bind()
                 Debug.Assert(_exprSrc == null || _typeSrc == _exprSrc.Type);    // type of source should be correct if source supplied
                 Debug.Assert(!_needsExprDest || _exprSrc != null);           // need source expr to create dest expr
 
+                Debug.Assert(_typeDest.GetTypeKind() != TypeKind.TK_MethodGroupType);
                 switch (_typeDest.GetTypeKind())
```

Comment:
With this change, in case of TK_MethodGroupType, it wouldn't return false. Is this by design?